
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for the Cancer Genomics workshop.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>Cohort analysis for the identification of driver genes - Cancer Genomics Workshop</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Cancer Genomics Workshop" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Hands-on Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Somatic Mutations
                </span>
              
            
            Cohort analysis for the identification of driver genes
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Cancer Genomics Workshop
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Hands-on Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Hands-on Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-qc/ngs-qc/" title="Quality Control" class="md-nav__link">
      Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-7" type="checkbox" id="nav-5-7" checked>
    
    <label class="md-nav__link" for="nav-5-7">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-7">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Cohort analysis for the identification of driver genes
      </label>
    
    <a href="./" title="Cohort analysis for the identification of driver genes" class="md-nav__link md-nav__link--active">
      Cohort analysis for the identification of driver genes
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" title="Author Information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysing-cancer-cohort-data-with-intogen" title="Analysing cancer cohort data with IntOGen" class="md-nav__link">
    Analysing cancer cohort data with IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exploring-the-output-of-intogen" title="Exploring the output of IntOGen" class="md-nav__link">
    Exploring the output of IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../license/" title="License" class="md-nav__link">
      License
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" title="Author Information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysing-cancer-cohort-data-with-intogen" title="Analysing cancer cohort data with IntOGen" class="md-nav__link">
    Analysing cancer cohort data with IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exploring-the-output-of-intogen" title="Exploring the output of IntOGen" class="md-nav__link">
    Exploring the output of IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Cohort analysis for the identification of driver genes</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Run the <code class="codehilite">IntOGen</code> analysis software on cohort mutation data.</p>
</li>
<li>
<p>Have gained experience of the structure of the analysis output files
    in order to identify potential driver genes.</p>
</li>
<li>
<p>Have gained overview knowledge of different methods for
    identification of genes important in cancers.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>IntOGen mutations platform:<br />
<a href="https://www.intogen.org/search">https://www.intogen.org/search</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p>TCGA melanoma somatic SNV data from 338 tumour samples:<br />
<a href="https://tcga-data.nci.nih.gov/tcga/">https://tcga-data.nci.nih.gov/tcga/</a></p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<p>Mutation Annotation Format (MAF) specification:<br />
<a href="https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification">https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification</a></p>
<p>IntOGen installation instructions:
<a href="https://bitbucket.org/intogen/intogen-pipeline/overview">https://bitbucket.org/intogen/intogen-pipeline/overview</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Ann-Marie Patch, QIMR Berghofer <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#97;&#110;&#110;&#45;&#109;&#97;&#114;&#105;&#101;&#46;&#112;&#97;&#116;&#99;&#104;&#64;&#113;&#105;&#109;&#114;&#98;&#101;&#114;&#103;&#104;&#111;&#102;&#101;&#114;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#97;&#110;&#110;&#45;&#109;&#97;&#114;&#105;&#101;&#46;&#112;&#97;&#116;&#99;&#104;&#64;&#113;&#105;&#109;&#114;&#98;&#101;&#114;&#103;&#104;&#111;&#102;&#101;&#114;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a><br />
Erdahl Teber, CMRI <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#101;&#116;&#101;&#98;&#101;&#114;&#64;&#99;&#109;&#114;&#105;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#101;&#116;&#101;&#98;&#101;&#114;&#64;&#99;&#109;&#114;&#105;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a>  </p>
<p><em>Contributor(s):</em><br />
Scott Wood <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#115;&#99;&#111;&#116;&#116;&#46;&#119;&#111;&#111;&#100;&#64;&#113;&#105;&#109;&#114;&#98;&#101;&#114;&#103;&#104;&#111;&#102;&#101;&#114;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#115;&#99;&#111;&#116;&#116;&#46;&#119;&#111;&#111;&#100;&#64;&#113;&#105;&#109;&#114;&#98;&#101;&#114;&#103;&#104;&#111;&#102;&#101;&#114;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a>  </p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>Cancer driver genes are commonly described as genes that when mutated
directly affect the potential of a cell to become cancerous. They are
important to a tumour cell as they confer a growth or survival advantage
over the normal surrounding cells. The mutations in these driver genes
are then clonally selected for as the population of tumour cells
increases. We think of the key genes driving tumour initiation
(development), progression, metastases, resistance and survival. Driver
gene mutations are often described as &ldquo;early&rdquo; events because they were
key in turning a normally functioning and regulated cell into a
dysregulated one. The logical assumption is that these key mutations
will be present in all tumour cells in a patient’s sample; although
sometime this is not true.</p>
<p>There are two major research goals that underline the need to identify
driver genes:</p>
<ul>
<li>
<p>By identifying the early changes that take place researchers might
    be able to find a treatment to stop the root cause of why cells
    become malignant.</p>
</li>
<li>
<p>By identifying groups of patients with the same genes mutated then
    we can develop therapies that will work for all of them.</p>
</li>
</ul>
<p>When we sequence tumour samples we tend to use samples that come from
fully developed cancers that can carry hundreds to thousands of
mutations in genes and many more outside of genes. The accumulation of
these passenger mutations in cancer cells can happen because often the
repair mechanisms or damage sensing processes are amongst the first
pathways to become disrupted accelerating the mutational rate. Mutations
that occur in genes after the cell has become cancerous may still affect
the growth rate, invasiveness and even the response to chemotherapy but
may not be present in all cells of a tumour. These genes may be drivers
of chemo-resistance or metastasis and are equally good targets for
therapies.</p>
<p>IntOGen-mutations is a platform that aims to identify driver mutations
using two methodologies from cancer cohort mutation data: the first
identifies mutations that are most likely to have a functional impact
combined with identifying genes that are frequently mutated; and the
second, genes that harbour clustered mutations. These measures are all
indicators of positive selection that occurs in cancer evolution and may
help the identification of driver genes.</p>
<hr />
<h2 id="analysing-cancer-cohort-data-with-intogen">Analysing cancer cohort data with IntOGen<a class="headerlink" href="#analysing-cancer-cohort-data-with-intogen" title="Permanent link">&para;</a></h2>
<p>IntOGen-mutations is available as a web based service that can allow
users to run their analysis on the host’s servers or it can be downloaded
and run on a local server.</p>
<p>For the purposes of the course we will be using a local version of
<code class="codehilite">IntOGen</code> so that we don’t encounter any issues sharing resources.</p>
<ul>
<li>
<p>To begin open a terminal and navigate to the directory <code class="codehilite">somatic/intogen</code>.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/somatic/intogen
</pre></div>
</td></tr></table>

</li>
</ul>
<p>In this directory you will find a Mutation Annotation Format (MAF) file
containing a cut down version of the somatic variant calls identified from
melanoma samples investigated as part of the TCGA cancer genomics projects.
You can see what files are in the directory by typing <code class="codehilite">ls</code>, look inside the
file using <code class="codehilite">less TCGA_Melanoma_SMgene.maf</code> and close the file and return to
the command line by typing <code class="codehilite">q</code>.</p>
<ul>
<li>
<p>Run the <code class="codehilite">IntOGen</code> analysis by typing</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>intogen -i TCGA_Melanoma_slimSMgene.maf -o TCGA_Mela_out
</pre></div>
</td></tr></table>

</li>
</ul>
<p>The TCGA melanoma maf used in this practical has been modified from the
original to reduce processing time and only contains data for the top
680 mutated genes.</p>
<p>The tool will take around 10 minutes to run and the progress will be
indicated by the logging lines printed to the terminal. Once complete
the output can be explored.</p>
<p>Whilst the tool is running we can explore the options we have used to
run <code class="codehilite">IntOGen</code>.</p>
<ul>
<li>
<p>To get a list of <code class="codehilite">IntOGen</code> options open up a new terminal</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>intogen --help
</pre></div>
</td></tr></table>

</li>
</ul>
<p>This command will list the running options that you can alter as command line
inputs or in a configuration file. We are using the default options for this run
so we didn&rsquo;t have to supply a configuration file and we only used <code class="codehilite">-i</code> to set
the input and <code class="codehilite">-o</code> to control the mane of the output directory.</p>
<ul>
<li>
<p>To look at the default options open up the configuration file by typing</p>
<p><code class="codehilite">bashless ~./intogen/task.confless ~./intogen/system.conf</code></p>
</li>
</ul>
<p>It is important to set the correct genome assembly in the <code class="codehilite">task.conf</code> to match
the one that you used as your reference when the variant were called. In our
<code class="codehilite">task.conf</code> this should be <code class="codehilite">hg19</code>.</p>
<hr />
<h2 id="exploring-the-output-of-intogen">Exploring the output of IntOGen<a class="headerlink" href="#exploring-the-output-of-intogen" title="Permanent link">&para;</a></h2>
<p>When you run your data over the web on the remote site there is a browse
facility that allows you to explore your data using the web version of
the database. Running <code class="codehilite">IntOGen</code> locally provides the same tabular
information but in a flat file format.</p>
<p>There should be 14 files generated from a successful run of this version of <code class="codehilite">IntOGen</code>:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gene.tsv
gene.oncodriveclust
pathway.recurrences
gene.oncodrivefm
sample_gene.impact
gene.recurrences                                
sample_variant+transcript.impact
summary.tsv
transcript.recurrences
TCGA_Melanoma_slimSMgene.smconfig
oncodrivefm-pathways-MA_SCORE.tsv
oncodrivefm-pathways-PPH2_SCORE.tsv  
oncodrivefm-pathways-SIFT_SCORE.tsv  
pathway.oncodrivefm
</pre></div>
</td></tr></table>

<p>View these files by using <code class="codehilite">ls</code> as below.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ls ~/somatic/intogen/TCGA_Mela_out/project/TCGA_Melanoma_slimSMgene/
</pre></div>
</td></tr></table>

<p>This practical will concentrate on the identification of driver genes so
we will look at the main output concerning genes.
The <code class="codehilite">gene.tsv</code> is the main gene centric output summary table.</p>
<ul>
<li>
<p>Open up the <code class="codehilite">gene.tsv</code> file in <code class="codehilite">LibreOffice</code> by double clicking on the icon on your
desktop.</p>
</li>
<li>
<p>Select the file tab and click on open.</p>
</li>
<li>
<p>Navigate to the results directory
<code class="codehilite">~/somatic/intogen/TCGA_Mela_out/project/TCGA_Melanoma_slimSMgene/</code></p>
</li>
<li>
<p>Double click on <code class="codehilite">gene.tsv</code>.</p>
</li>
<li>
<p>In the pop-up box under the <code class="codehilite">Separator options</code> ensure only the tab box is
checked and click <code class="codehilite">OK</code>.</p>
</li>
</ul>
<p>This file contains the overall summary results for the <code class="codehilite">IntOGen</code> pipeline
presented by gene and reports Q values (i.e. multiple testing corrected
P values) for the mutation frequency and cluster modules.</p>
<p>Significantly mutated genes from the cohort data are identified using both
the <code class="codehilite">OncodriveFM</code> and <code class="codehilite">OncodriveClUST</code> modules of <code class="codehilite">IntOGen</code>. The <code class="codehilite">OncodriveFM</code>
module detects genes that have accumulated mutations with a high functional
impact. It uses annotations from the Ensembl variant effect predictor (VEP, V.70)
that includes SIFT and Polyphen2 and precomputed MutationAssessor functional
impacts. It calculates a P value per gene from the number of mutations detected
across all possible coding bases of a gene with a positive weighting for mutations
with a high functional impact. The <code class="codehilite">OncodriveCLUST</code> module detects genes that
have more variants than would be expected across the cohort that alter the
same region of the gene.</p>
<p>The file is sorted to bring the most significantly altered genes to the top. The key
columns that help you identify the significantly mutated genes are the 3<sup>rd</sup> and 4<sup>th</sup>
(C and D) that indicate which of the modules identified a significant result and
the Q-values for the modules that are in 21<sup>st</sup> and 23<sup>rd</sup> (U and W)</p>
<p>The top twelve genes have significant Q-values for both modules and include BRAF,
NRAS and TP53. The next 35 are significant by only one of the modules.</p>
<p>All of these have small Q-values which means they are all
significantly mutated genes in this TCGA Melanoma cohort of 338
patients.</p>
<ul>
<li>Now look at their sample frequency count (column 9 <code class="codehilite">MUTS_CS_SAMPLES</code>) these
are the number of samples that contain at least one mutation in the gene.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>a)   Which significantly mutated gene has mutations in the most samples?</p>
<p>b)   Which gene/genes have the lowest Q-value from OncodriveFM and OncodriveCLUST?</p>
<p>c)   Why don’t the genes with the lowest Q values also have the highest sample frequency value?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>a)  BRAF has 175 out of 327 cases with a mutation.</p><p>b)  TP53 or PTEN have the lowest OncodriveFM Q-values and NRAS
has the lowest Q-value for OncodriveCLUST.</p><p>c)  The P value calculation takes into account the length of the
    coding sequence of the gene, the mutation rate of the nucleotides
    contained within it and for OncodriveFM the functional consequences
    of those changes. Therefore a small gene with a small number of deleterious
    mutations may have a lower P value and also Q value than a large
    gene with a high mutation frequency.</p></details></div>
</div>
<p>The results for the assessment of clustered mutations in genes carried
out by the <code class="codehilite">OncodriveCLUST</code> module of <code class="codehilite">IntOGen</code> are shown as  amino
acid residue positions of the encoded protein.</p>
<p>The three known oncogenes BRAF, NRAS and IDH1 have very low CLUST_QVALUEs
indicating that the mutations in these genes are highly clustered. The
<code class="codehilite">CLUST_COORDS</code> column reports that there are 160 samples with mutations
between the amino acid positions 594-601 of BRAF; 84 samples with mutations
at amino acid position 61 of NRAS; and 15 sample with mutations at amino
acid position 132 of IDH1.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why are the oncogenes more likely to have clustered mutations and the
tumour suppressor genes less likely?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Gain of function mutations are required to activate oncogenes and so
only key residues in the protein will result in activation. Tumour
suppressors are frequently affected by loss of function mutations and
deletions. A truncating mutation or frameshift indel can occur in any
exon, except the last one, and have the same deleterious functional
result.</p></details></div>
</div>
<p><br>
The other files in the output support the information in this sheet.</p>
<p>The <code class="codehilite">sample_variant+transcript.impact</code> file includes a summary of all mutations
found in each of the genes and protein coding transcripts of those genes for all
samples identified that have that mutation. It also reports the variant impact scores
from SIFT, PolyPhen2, MutationAssesor, reporting also impact categories of which
there are four; high, medium, low and none.</p>
<ul>
<li>Open up the <code class="codehilite">sample_variant+transcript.impact</code> file and explore the data.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you find out what the nucleotide change details for the most common
BRAF mutation that results in V600E amino acid change in the cohort?
Sort the data by GENE, then TRANSCRIPT and then PROTEIN_POS to make this easier. The gene ID for BRAF is <code class="codehilite">ENSG00000157764</code>.</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>It is an A&gt;T at position chr7:140453136 identified in 127 samples.</p></details></div>
</div>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<p>Gunes et al. Nat. Methods 2010
:   <a href="http://www.nature.com/nmeth/journal/v7/n2/pdf/nmeth0210-92.pdf">http://www.nature.com/nmeth/journal/v7/n2/pdf/nmeth0210-92.pdf</a></p>
<p>Gonzalez-Perez et al. Nat. Methods 2013
:   <a href="http://www.nature.com/nmeth/journal/v10/n11/pdf/nmeth.2642.pdf">http://www.nature.com/nmeth/journal/v10/n11/pdf/nmeth.2642.pdf</a></p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../01_signatures/" title="Assessing somatic mutational signatures" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Assessing somatic mutational signatures
              </span>
            </div>
          </a>
        
        
          <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Variant Visualisation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>