
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for the Cancer Genomics workshop.">
      
      
        <meta name="author" content="Authors">
      
      
      
        <link rel="prev" href="../../cancer-module-alignment/alignment/">
      
      
        <link rel="next" href="../../cancer-module-cnv/cnv-tut/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Single Nucleotide Variant Calling and Annotation - Cancer Genomics Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#key-learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Cancer Genomics Workshop" class="md-header__button md-logo" aria-label="Cancer Genomics Workshop" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Cancer Genomics Workshop
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Single Nucleotide Variant Calling and Annotation
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Cancer Genomics Workshop" class="md-nav__button md-logo" aria-label="Cancer Genomics Workshop" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Cancer Genomics Workshop
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../trainers/trainers/" class="md-nav__link">
        The Trainers
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../timetables/timetable_cancer/" class="md-nav__link">
        Timetable
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../preamble/" class="md-nav__link">
        Workshop Information
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Hands-on Modules
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Hands-on Modules" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Hands-on Modules
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cli/commandline/" class="md-nav__link">
        Introduction to Command Line
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-qc/ngs-qc/" class="md-nav__link">
        Quality Control
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-alignment/alignment/" class="md-nav__link">
        Read Alignment
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Single Nucleotide Variant Calling and Annotation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Single Nucleotide Variant Calling and Annotation
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-files" class="md-nav__link">
    BAM Files
  </a>
  
    <nav class="md-nav" aria-label="BAM Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#exploring-bam-files" class="md-nav__link">
    Exploring BAM files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-1-pre-processing-indel-realignment" class="md-nav__link">
    Step 1: Pre-processing: Indel Realignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-pre-processing-fixmates" class="md-nav__link">
    Step 2: Pre-processing: Fixmates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-pre-processing-mark-duplicates" class="md-nav__link">
    Step 3: Pre-processing: Mark Duplicates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-base-quality-recalibration" class="md-nav__link">
    Step 4: Pre-processing: Base Quality Recalibration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-qc" class="md-nav__link">
    BAM QC
  </a>
  
    <nav class="md-nav" aria-label="BAM QC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-bam-qc-compute-coverage" class="md-nav__link">
    Step 1: BAM QC: Compute Coverage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-bam-qc-insert-size" class="md-nav__link">
    Step 2: BAM QC: Insert Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-bam-qc-alignment-metrics" class="md-nav__link">
    Step 3: BAM QC: Alignment metrics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" class="md-nav__link">
    Variant Calling
  </a>
  
    <nav class="md-nav" aria-label="Variant Calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#varscan-2" class="md-nav__link">
    Varscan 2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mutect" class="md-nav__link">
    MuTect
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#comparing-variant-callers" class="md-nav__link">
    Comparing variant callers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-visualisation" class="md-nav__link">
    Variant Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-annotation" class="md-nav__link">
    Variant Annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-filtration" class="md-nav__link">
    Variant Filtration
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-cnv/cnv-tut/" class="md-nav__link">
        Copy Number Variation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-sv/sv_tut/" class="md-nav__link">
        Structural Variant Analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_7" type="checkbox" id="__nav_5_7" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5_7">
          Somatic Mutations
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Somatic Mutations" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_7">
          <span class="md-nav__icon md-icon"></span>
          Somatic Mutations
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/01_signatures/" class="md-nav__link">
        Assessing somatic mutational signatures
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-somatic/02_intogen/" class="md-nav__link">
        Cohort analysis for the identification of driver genes
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cancer-module-viz/visu/" class="md-nav__link">
        Variant Visualisation
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../license/" class="md-nav__link">
        License
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav" aria-label="Resources You’ll be Using">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-files" class="md-nav__link">
    BAM Files
  </a>
  
    <nav class="md-nav" aria-label="BAM Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#exploring-bam-files" class="md-nav__link">
    Exploring BAM files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-1-pre-processing-indel-realignment" class="md-nav__link">
    Step 1: Pre-processing: Indel Realignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-pre-processing-fixmates" class="md-nav__link">
    Step 2: Pre-processing: Fixmates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-pre-processing-mark-duplicates" class="md-nav__link">
    Step 3: Pre-processing: Mark Duplicates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-base-quality-recalibration" class="md-nav__link">
    Step 4: Pre-processing: Base Quality Recalibration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-qc" class="md-nav__link">
    BAM QC
  </a>
  
    <nav class="md-nav" aria-label="BAM QC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-bam-qc-compute-coverage" class="md-nav__link">
    Step 1: BAM QC: Compute Coverage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-bam-qc-insert-size" class="md-nav__link">
    Step 2: BAM QC: Insert Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-bam-qc-alignment-metrics" class="md-nav__link">
    Step 3: BAM QC: Alignment metrics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" class="md-nav__link">
    Variant Calling
  </a>
  
    <nav class="md-nav" aria-label="Variant Calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#varscan-2" class="md-nav__link">
    Varscan 2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mutect" class="md-nav__link">
    MuTect
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#strelka" class="md-nav__link">
    Strelka
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#comparing-variant-callers" class="md-nav__link">
    Comparing variant callers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-visualisation" class="md-nav__link">
    Variant Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-annotation" class="md-nav__link">
    Variant Annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-filtration" class="md-nav__link">
    Variant Filtration
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" class="md-nav__link">
    References
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Single Nucleotide Variant Calling and Annotation</h1>

<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Prepare raw BAM alignments for variant detection</p>
</li>
<li>
<p>Perform QC measures on BAM files</p>
</li>
<li>
<p>Perform simple variant detection on paired NGS data</p>
</li>
<li>
<p>Add annotation information to raw variant calls</p>
</li>
<li>
<p>Visualise variant calls using IGV</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>SAMTools:<br />
<a href="https://samtools.github.io/">https://samtools.github.io/</a></p>
<p>IGV:<br />
<a href="http://www.broadinstitute.org/igv/">http://www.broadinstitute.org/igv/</a></p>
<p>Genome Analysis Toolkit:<br />
<a href="http://www.broadinstitute.org/gatk/">http://www.broadinstitute.org/gatk/</a></p>
<p>Picard:<br />
<a href="https://broadinstitute.github.io/picard/">https://broadinstitute.github.io/picard/</a></p>
<p>MuTect:<br />
<a href="http://www.broadinstitute.org/cancer/cga/mutect/">http://www.broadinstitute.org/cancer/cga/mutect/</a></p>
<p>Strelka:<br />
<a href="https://sites.google.com/site/strelkasomaticvariantcaller/">https://sites.google.com/site/strelkasomaticvariantcaller/</a></p>
<p>VarScan2:<br />
<a href="https://dkoboldt.github.io/varscan/">https://dkoboldt.github.io/varscan/</a></p>
<p>Variant Effect Predictor:<br />
<a href="http://www.ensembl.org/info/docs/tools/vep">http://www.ensembl.org/info/docs/tools/vep</a></p>
<p>GEMINI:<br />
<a href="http://gemini.readthedocs.org">http://gemini.readthedocs.org</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p><a href="http://sra.dnanexus.com/studies/ERP001071">http://sra.dnanexus.com/studies/ERP001071</a><br />
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22194472">http://www.ncbi.nlm.nih.gov/pubmed/22194472</a></p>
<hr />
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p><em>Primary Author(s):</em><br />
Matt Field <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#109;&#97;&#116;&#116;&#46;&#102;&#105;&#101;&#108;&#100;&#64;&#97;&#110;&#117;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#109;&#97;&#116;&#116;&#46;&#102;&#105;&#101;&#108;&#100;&#64;&#97;&#110;&#117;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a><br />
Dan Andrews <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#100;&#97;&#110;&#46;&#97;&#110;&#100;&#114;&#101;&#119;&#115;&#64;&#97;&#110;&#117;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#100;&#97;&#110;&#46;&#97;&#110;&#100;&#114;&#101;&#119;&#115;&#64;&#97;&#110;&#117;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a><br />
Velimir Gayevskiy <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#118;&#46;&#103;&#97;&#121;&#101;&#118;&#115;&#107;&#105;&#121;&#64;&#103;&#97;&#114;&#118;&#97;&#110;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#118;&#46;&#103;&#97;&#121;&#101;&#118;&#115;&#107;&#105;&#121;&#64;&#103;&#97;&#114;&#118;&#97;&#110;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a><br />
Mathieu Bourgey <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#109;&#97;&#116;&#104;&#105;&#101;&#117;&#46;&#98;&#111;&#117;&#114;&#103;&#101;&#121;&#64;&#109;&#99;&#103;&#105;&#108;&#108;&#46;&#99;&#97;">&#109;&#97;&#116;&#104;&#105;&#101;&#117;&#46;&#98;&#111;&#117;&#114;&#103;&#101;&#121;&#64;&#109;&#99;&#103;&#105;&#108;&#108;&#46;&#99;&#97;</a>  </p>
<p><em>Contributor(s):</em><br />
Gayle Philip <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#83;&#111;&#110;&#105;&#107;&#97;&#46;&#84;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;">&#83;&#111;&#110;&#105;&#107;&#97;&#46;&#84;&#121;&#97;&#103;&#105;&#64;&#97;&#103;&#114;&#102;&#46;&#111;&#114;&#103;&#46;&#97;&#117;</a><br />
Sonika Tyagi <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#103;&#107;&#112;&#104;&#105;&#108;&#105;&#112;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#103;&#107;&#112;&#104;&#105;&#108;&#105;&#112;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a>  </p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>The goal of this hands-on session is to present the main steps that are
commonly used to process and to analyze cancer sequencing data. We will
focus only on whole genome data and provide command lines that allow
detecting Single Nucleotide Variants (SNV). This workshop will show you
how to launch individual steps of a complete DNA-Seq SNV pipeline using
cancer data.</p>
<p>In the second part of the tutorial we will also be using IGV to
visualise and manually inspect candidate variant calls.</p>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>We will use a dataset derived from whole genome sequencing of a
33-yr-old lung adenocarcinoma patient, who is a never-smoker and has no
familial cancer history.</p>
<p>The data consists of whole genome sequencing of liver metastatic lung
cancer (frozen), primary lung cancer (FFPE) and blood tissue of a lung
adenocarcinoma patient (AK55).</p>
<p>Open the Terminal and go to the <code>snv</code> working directory:</p>
<div class="codehilite"><pre><span></span><code>cd /home/trainee/snv/
</code></pre></div>

<div class="admonition failure">
<p>All commands entered into the terminal for this tutorial should be from
within the <strong><code>snv</code></strong> directory.</p>
</div>
<p>The BAM alignment files are contained in the subdirectory called
<code>alignment</code> and are located in the following subdirectories:</p>
<blockquote>
<p><code>normal/normal.sorted.bam</code> and <code>normal/normal.sorted.bam.bai</code></p>
<p><code>tumour/tumor.sorted.bam</code> and <code>tumour/tumour.sorted.bam.bai</code></p>
</blockquote>
<p>Check that the <code>alignment</code> directory contains the above-mentioned files by typing:</p>
<div class="codehilite"><pre><span></span><code>ls -l alignment/*
</code></pre></div>

<p><br>
These files are based on subsetting the whole genomes derived from blood
and liver metastases to the first 10Mb of chromosome 4. This will allow
our analyses to run in a sufficient time during the workshop, but it’s
worth being aware that this is less &lt; 0.5% of the genome which
highlights the length of time and resources required to perform cancer
genomics on full genomes!</p>
<p>The initial structure of your folders should look like this (type <code>ls -l</code>):</p>
<div class="highlight"><pre><span></span><code>-- alignment/             # bam files
  -- normal/                # The blood sample directory containing bam files
  -- tumour/                # The tumour sample directory containing bam files
-- ref/                   # Contains reference genome files      
</code></pre></div>
<p><br>
Now we need to set some environment variables to save typing lengthy
file paths over and over. Copy and paste the following commands into
your terminal.</p>
<div class="highlight"><pre><span></span><code><span class="nb">export</span><span class="w"> </span><span class="nv">APP_ROOT</span><span class="o">=</span>/home/trainee/snv/Applications
<span class="nb">export</span><span class="w"> </span><span class="nv">IGVTOOLS_PATH</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/igvtools/
<span class="nb">export</span><span class="w"> </span><span class="nv">PICARD_JAR</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/picard/picard.jar
<span class="nb">export</span><span class="w"> </span><span class="nv">GATK_JAR</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/gatk/GenomeAnalysisTK.jar
<span class="nb">export</span><span class="w"> </span><span class="nv">STRELKA_HOME</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/strelka/
<span class="nb">export</span><span class="w"> </span><span class="nv">MUTECT_JAR</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/mutect/muTect-1.1.5.jar
<span class="nb">export</span><span class="w"> </span><span class="nv">VARSCAN_JAR</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/varscan/VarScan.v2.4.1.jar
<span class="nb">export</span><span class="w"> </span><span class="nv">REF</span><span class="o">=</span>/home/trainee/snv/ref
<span class="nb">export</span><span class="w"> </span><span class="nv">SNV_BASE</span><span class="o">=</span>/home/trainee/snv
<span class="nb">export</span><span class="w"> </span><span class="nv">JAVA7</span><span class="o">=</span>/usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java
<span class="nb">export</span><span class="w"> </span><span class="nv">IGV</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/igv/igv.sh
<span class="nb">export</span><span class="w"> </span><span class="nv">VEP</span><span class="o">=</span><span class="nv">$APP_ROOT</span>/ensembl-tools/scripts/variant_effect_predictor/variant_effect_predictor.pl
<span class="nb">export</span><span class="w"> </span><span class="nv">VEP_CACHE</span><span class="o">=</span>/mnt/workshop/data/bgdata/datasets/vepcache/82
</code></pre></div>
<p><br>
Make sure you are in the correct directory by typing:</p>
<div class="codehilite"><pre><span></span><code>cd $SNV_BASE
</code></pre></div>

<hr />
<h2 id="bam-files">BAM Files<a class="headerlink" href="#bam-files" title="Permanent link">&para;</a></h2>
<p>Let’s spend some time exploring BAM files.</p>
<h3 id="exploring-bam-files">Exploring BAM files<a class="headerlink" href="#exploring-bam-files" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span><code><span class="n">samtools</span><span class="w"> </span><span class="n">view</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">head</span><span class="w"> </span><span class="o">-</span><span class="n">n4</span>
</code></pre></div>

<p>Here you have examples of alignment results. A full description of the
flags can be found in the <a href="http://samtools.sourceforge.net/SAM1.pdf">SAM specification</a>.</p>
<p>Another useful bit of information in the SAM is the CIGAR string. It’s
the 6<sup>th</sup> column in the file.</p>
<p>This column explains how the alignment was achieved.</p>
<blockquote>
<p>M == base aligns <strong>but doesn’t have to be a match.</strong> A SNP will have an M even if it disagrees with the reference.<br />
I == Insertion<br />
D == Deletion<br />
S == soft-clips. These are handy to find un removed adapters, viral insertions, etc.</p>
</blockquote>
<p>An in-depth explanation of the CIGAR can be found <a href="http://genome.sph.umich.edu/wiki/SAM">here</a>.
The exact details of the CIGAR string can be found in the SAM specification as
well. We won’t go into too much detail at this point since we want to concentrate on
cancer specific issues now.</p>
<p>Now, you can try using Picard&rsquo;s <a href="http://broadinstitute.github.io/picard/explain-flags.html">explain flag</a>
site to understand what is going on with your reads.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>There are 3 unique flags, what do they mean? The flag is the second column.</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p><strong>129:</strong><br />
read paired<br />
second in pair  </p>
<p><strong>113:</strong><br />
read paired<br />
read reverse strand<br />
mate reverse strand<br />
first in pair  </p>
<p><strong>161:</strong><br />
read paired<br />
mate reverse strand<br />
second in pair  </p>
</details>
</div>
</div>
<p><br>
There are lots of possible different flags, let’s look at a few more</p>
<div class="codehilite"><pre><span></span><code><span class="n">samtools</span><span class="w"> </span><span class="n">view</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">head</span><span class="w"> </span><span class="o">-</span><span class="n">n</span><span class="w"> </span><span class="mi">100</span>
</code></pre></div>

<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Let’s take the last read, which looks properly paired and find its mate pair.</p>
<div class="admonition hint">
<details>
<summary><strong>Hint</strong></summary>
<p>a)    Instead of using <code>head</code>, what unix command could we pipe the output to?<br />
b)    Once we’ve found both reads, the command can be stopped by typing <code>CTRL-C</code>  </p>
</details>
</div>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<div class="codehilite"><pre><span></span><code><span class="n">samtools</span><span class="w"> </span><span class="n">view</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">grep</span><span class="w"> </span><span class="n">HWI</span><span class="o">-</span><span class="n">ST478_0133</span><span class="p">:</span><span class="mi">4</span><span class="p">:</span><span class="mi">2205</span><span class="p">:</span><span class="mi">14675</span><span class="p">:</span><span class="mi">32513</span>
</code></pre></div>

</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Using the cigar string, what can we tell about the alignment of the mate pair?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The mate pair has a less convincing alignment with two insertions and
soft clipping reported.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How might the alignment information from the original read be used by the aligner?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Even though the alignment of the mate pair is questionable the presence
  of it’s properly paired mate helps the aligner in deciding where to put
  the less-certain read.</p>
</details>
</div>
</div>
<p>You can use <code>Samtools</code> to filter reads as well.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How many reads mapped and unmapped were there?</p>
<div class="admonition hint">
<details>
<summary><strong>Hint</strong></summary>
<p>Look at the samtools view help menu by typing <code>samtools view</code> without any arguments</p>
</details>
</div>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p><div class="highlight"><pre><span></span><code>samtools view -c -f4 alignment/normal/normal.sorted.bam
</code></pre></div>
77229
<div class="highlight"><pre><span></span><code>samtools view -c -F4 alignment/normal/normal.sorted.bam
</code></pre></div>
22972373</p>
</details>
</div>
</div>
<h3 id="step-1-pre-processing-indel-realignment">Step 1: Pre-processing: Indel Realignment<a class="headerlink" href="#step-1-pre-processing-indel-realignment" title="Permanent link">&para;</a></h3>
<p>The first step for this is to realign around indels and SNP dense regions.
The Genome Analysis toolkit (<code>GATK</code>) has a tool for this called IndelRealigner.
It basically runs in 2 steps:  </p>
<ol>
<li>Find the targets</li>
<li>Realign them</li>
</ol>
<div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w">  </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">GATK_JAR</span><span class="p">}</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">T</span><span class="w"> </span><span class="n">RealignerTargetCreator</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">R</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">human_g1k_v37</span><span class="p">.</span><span class="na">fasta</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">o</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">realign</span><span class="p">.</span><span class="na">intervals</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">I</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">I</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">L</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">human_g1k_v37</span><span class="p">.</span><span class="na">intervals</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">GATK_JAR</span><span class="p">}</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">T</span><span class="w"> </span><span class="n">IndelRealigner</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">R</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">human_g1k_v37</span><span class="p">.</span><span class="na">fasta</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">targetIntervals</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">realign</span><span class="p">.</span><span class="na">intervals</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">nWayOut</span><span class="w"> </span><span class="p">.</span><span class="na">realigned</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">I</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">I</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">L</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">human_g1k_v37</span><span class="p">.</span><span class="na">intervals</span>
</code></pre></div>
<p>Explanation of parameters:</p>
<blockquote>
<p><strong>-I</strong>: BAM file(s)<br />
<strong>-T</strong>: GATK algorithm to run<br />
<strong>-R</strong>: the reference genome used for mapping (b37 from GATK here)<br />
<strong>-jar</strong>: Path to GATK jar file<br />
<strong>-L</strong>: Genomic intervals to operate on</p>
</blockquote>
<p><br>
Move the realigned BAMs and index files to the corresponding normal and
tumour directories.</p>
<div class="highlight"><pre><span></span><code>mv<span class="w"> </span>normal.sorted.realigned.ba*<span class="w"> </span>alignment/normal/
mv<span class="w"> </span>tumour.sorted.realigned.ba*<span class="w"> </span>alignment/tumour/
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why did we use both normal and tumor together?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Because if a region needs realignment, maybe one of the samples in the
pair has less reads or was excluded from the target creation.
This makes sure the normal and tumor are all in-sync for the somatic
calling step.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How many regions did it think needed cleaning?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<div class="codehilite"><pre><span></span><code>wc -l alignment/normal/realign.intervals
</code></pre></div>

<p>27300</p>
</details>
</div>
</div>
<p><br>
Indel Realigner also makes sure the called deletions are left aligned
when there is a microsatellite or homopolymer. e.g.</p>
<p>This<br />
  ATCGAAAA-TCG<br />
  into<br />
  ATCG-AAAATCG  </p>
<p>or  </p>
<p>ATCGATATATATA&ndash;TCG<br />
  into<br />
  ATCG&ndash;ATATATATATCG  </p>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why is it important?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>This makes it easier for downstream analysis tools.</p>
<p>For NGS analysis, the convention is to left align indels.</p>
<p>This is only really needed when calling variants with legacy locus-based
tools such as samtools or GATK UnifiedGenotyper. Otherwise you will have
worse performance and accuracy.</p>
<p>With more sophisticated tools (like GATK HaplotypeCaller) that involve
reconstructing haplotypes (e.g. through reassembly), the problem of
multiple valid representations is handled internally and does not need
to be corrected explicitly.</p>
</details>
</div>
</div>
<h3 id="step-2-pre-processing-fixmates">Step 2: Pre-processing: Fixmates<a class="headerlink" href="#step-2-pre-processing-fixmates" title="Permanent link">&para;</a></h3>
<p>Some read entries don’t have their mate information written properly.<br />
We use <code>Picard</code> to do this:</p>
<p>Normal sample:
  <div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">PICARD_JAR</span><span class="p">}</span><span class="w"> </span><span class="n">FixMateInformation</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">VALIDATION_STRINGENCY</span><span class="o">=</span><span class="n">SILENT</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_INDEX</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">SORT_ORDER</span><span class="o">=</span><span class="n">coordinate</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">MAX_RECORDS_IN_RAM</span><span class="o">=</span><span class="mi">500000</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">INPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">realigned</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">OUTPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">matefixed</span><span class="p">.</span><span class="na">bam</span>
</code></pre></div></p>
<p>Tumour sample: <br />
  <div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">PICARD_JAR</span><span class="p">}</span><span class="w"> </span><span class="n">FixMateInformation</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">VALIDATION_STRINGENCY</span><span class="o">=</span><span class="n">SILENT</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_INDEX</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">SORT_ORDER</span><span class="o">=</span><span class="n">coordinate</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">MAX_RECORDS_IN_RAM</span><span class="o">=</span><span class="mi">500000</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">INPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">realigned</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">OUTPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">matefixed</span><span class="p">.</span><span class="na">bam</span>
</code></pre></div></p>
<h3 id="step-3-pre-processing-mark-duplicates">Step 3: Pre-processing: Mark Duplicates<a class="headerlink" href="#step-3-pre-processing-mark-duplicates" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are duplicate reads?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Different read pairs representing the same initial DNA fragment.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are they caused by?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ul>
<li>PCR reactions (PCR duplicates).  </li>
<li>Some clusters that are thought of being separate in the flowcell but are
the same (optical duplicates)</li>
</ul>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are the ways to detect them?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ol>
<li>
<p>Picard and samtools uses the alignment positions:  </p>
<ul>
<li>Both 5’ ends of both reads need to have the same positions.</li>
<li>Each reads have to be on the same strand as well.</li>
</ul>
</li>
<li>
<p>Another method is to use a kmer approach:</p>
<ul>
<li>Take a part of both ends of the fragment.</li>
<li>Build a hash table.</li>
<li>Count the similar hits.</li>
</ul>
</li>
<li>
<p>Brute force, compare all the sequences.</p>
</li>
</ol>
</details>
</div>
</div>
<p><br>
Here we will use <code>Picard</code>&lsquo;s approach:</p>
<p>Normal Sample:
  <div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">PICARD_JAR</span><span class="p">}</span><span class="w"> </span><span class="n">MarkDuplicates</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">REMOVE_DUPLICATES</span><span class="o">=</span><span class="kc">false</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_MD5_FILE</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">VALIDATION_STRINGENCY</span><span class="o">=</span><span class="n">SILENT</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_INDEX</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">INPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">matefixed</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">OUTPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">METRICS_FILE</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">metrics</span>
</code></pre></div></p>
<p>Tumour Sample:<br />
  <div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">PICARD_JAR</span><span class="p">}</span><span class="w"> </span><span class="n">MarkDuplicates</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">REMOVE_DUPLICATES</span><span class="o">=</span><span class="kc">false</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_MD5_FILE</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">VALIDATION_STRINGENCY</span><span class="o">=</span><span class="n">SILENT</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">CREATE_INDEX</span><span class="o">=</span><span class="kc">true</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">INPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">matefixed</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">OUTPUT</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="n">METRICS_FILE</span><span class="o">=</span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">metrics</span>
</code></pre></div></p>
<p><br>
We can look in the metrics output to see what happened.</p>
<div class="codehilite"><pre><span></span><code>less alignment/normal/normal.sorted.dup.metrics
</code></pre></div>

<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What percent of reads are duplicates?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>0.046996%</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Often, we have multiple libraries and when this occurs separate measures
are calculated for each library. Why is it important to not combine everything?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ul>
<li>Each library represents a set of different DNA fragments.</li>
<li>Each library involves different PCR reactions</li>
</ul>
<p>PCR duplicates can not occur between fragments of two different
libraries. However, similar fragments could be found between
libraries when the coverage is high.</p>
</details>
</div>
</div>
<h3 id="step-4-pre-processing-base-quality-recalibration">Step 4: Pre-processing: Base Quality Recalibration<a class="headerlink" href="#step-4-pre-processing-base-quality-recalibration" title="Permanent link">&para;</a></h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why do we need to recalibrate base quality scores?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The vendors tend to inflate the values of the bases in the reads. The
recalibration tries to lower the scores of some biased motifs for some
technologies.</p>
</details>
</div>
</div>
<p><br>
Base Quality Recalibration runs in 2 steps:</p>
<ol>
<li>Build covariates based on context and known SNP sites.</li>
<li>Correct the reads based on these metrics.</li>
</ol>
<p>GATK BaseRecalibrator:</p>
<div class="codehilite"><pre><span></span><code><span class="k">for</span> <span class="n">i</span> <span class="nb">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="nb">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
    -<span class="n">T</span> <span class="n">BaseRecalibrator</span> \
    -<span class="n">nct</span> <span class="mi">2</span> \
    -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    -<span class="n">knownSites</span> ${<span class="n">REF</span>}/<span class="n">dbSnp-138_chr4</span>.<span class="n">vcf</span> \
    -<span class="n">L</span> <span class="mi">4</span>:<span class="mi">1</span><span class="o">-</span><span class="mi">10000000</span> \
    -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recalibration_report</span>.<span class="n">grp</span> \
    -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">bam</span>

    <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
      -<span class="n">T</span> <span class="n">PrintReads</span> \
      -<span class="n">nct</span> <span class="mi">2</span> \
      -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
      -<span class="n">BQSR</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recalibration_report</span>.<span class="n">grp</span> \
      -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
      -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">bam</span>
<span class="nb">done</span>
</code></pre></div>

<hr />
<h2 id="bam-qc">BAM QC<a class="headerlink" href="#bam-qc" title="Permanent link">&para;</a></h2>
<p>Once your whole BAM is generated, it’s always a good thing to check the
data again to see if everything makes sense.</p>
<h3 id="step-1-bam-qc-compute-coverage">Step 1: BAM QC: Compute Coverage<a class="headerlink" href="#step-1-bam-qc-compute-coverage" title="Permanent link">&para;</a></h3>
<p>If you have data from a capture kit, you should see how well your
targets worked. <code>GATK</code> has a depth of coverage tool to do this:</p>
<div class="codehilite"><pre><span></span><code><span class="k">for</span> <span class="n">i</span> <span class="nb">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="nb">do</span>
  <span class="nv">$JAVA7</span>  -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
    -<span class="n">T</span> <span class="n">DepthOfCoverage</span> \
    --<span class="n">omitDepthOutputAtEachBase</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">10</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">25</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">50</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">100</span> \
    --<span class="n">start</span> <span class="mi">1</span> --<span class="n">stop</span> <span class="mi">500</span> --<span class="n">nBins</span> <span class="mi">499</span> -<span class="n">dt</span> <span class="n">NONE</span> \
    -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">coverage</span> \
    -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    -<span class="n">L</span> <span class="mi">4</span>:<span class="mi">1</span><span class="o">-</span><span class="mi">10000000</span>
<span class="nb">done</span>
</code></pre></div>

<p>Explanation of parameters:</p>
<blockquote>
<p><strong>--omitBaseOutputAtEachBase</strong>: Do not output depth of coverage at each base<br />
<strong>--summaryCoverageThreshold</strong>: Coverage threshold (in percent) for summarizing statistics<br />
<strong>-dt</strong>: Downsampling<br />
<strong>-L</strong>: Genomic intervals to operate on  </p>
</blockquote>
<p><br>
In this project, coverage is expected to be 25x. Look at the coverage:</p>
<div class="codehilite"><pre><span></span><code>less -S alignment/normal/normal.sorted.dup.recal.coverage.sample_interval_summary
</code></pre></div>

<p>Type <code>q</code> to return to the prompt.</p>
<div class="codehilite"><pre><span></span><code>less -S alignment/tumour/tumour.sorted.dup.recal.coverage.sample_interval_summary
</code></pre></div>

<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Does the coverage fit with the expectation?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<ul>
<li>
<p>Yes the mean coverage of the region is 25x.</p>
</li>
<li>
<p><code>summaryCoverageThreshold</code> is a useful function to see if your coverage is
uniform.</p>
</li>
<li>
<p>Another way is to compare the mean to the median. If both are quite
different that means something is wrong in your coverage.</p>
</li>
</ul>
</details>
</div>
</div>
<h3 id="step-2-bam-qc-insert-size">Step 2: BAM QC: Insert Size<a class="headerlink" href="#step-2-bam-qc-insert-size" title="Permanent link">&para;</a></h3>
<p>Insert size corresponds to the size of DNA fragments sequenced. It is different
from the gap size (= distance between reads)!</p>
<p><br>
These metrics are computed using <code>Picard</code>:</p>
<div class="codehilite"><pre><span></span><code><span class="k">for</span> <span class="n">i</span> <span class="nb">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="nb">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">PICARD_JAR</span>} <span class="n">CollectInsertSizeMetrics</span> \
    <span class="n">VALIDATION_STRINGENCY</span>=<span class="n">SILENT</span> \
    <span class="n">REFERENCE_SEQUENCE</span>=${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    <span class="n">INPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">OUTPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">insertSize</span>.<span class="n">tsv</span> \
    <span class="n">HISTOGRAM_FILE</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">insertSize</span>.<span class="n">histo</span>.<span class="n">pdf</span> \
    <span class="n">METRIC_ACCUMULATION_LEVEL</span>=<span class="n">LIBRARY</span>
<span class="nb">done</span>
</code></pre></div>

<p><br>
Look at the output:</p>
<div class="codehilite"><pre><span></span><code>less -S alignment/normal/normal.sorted.dup.recal.metric.insertSize.tsv
less -S alignment/tumour/tumour.sorted.dup.recal.metric.insertSize.tsv
</code></pre></div>

<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do the two libraries compare?    </p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>The tumour sample has a larger median insert size than the
normal sample (405 vs. 329).</p>
</details>
</div>
</div>
<h3 id="step-3-bam-qc-alignment-metrics">Step 3: BAM QC: Alignment metrics<a class="headerlink" href="#step-3-bam-qc-alignment-metrics" title="Permanent link">&para;</a></h3>
<p>Alignment metrics tells you if your sample and your reference fit together.</p>
<p>For the alignment metrics, <code>samtools flagstat</code> is very fast but
<code>bwa-mem</code> breaks some reads into pieces, the numbers can be a bit confusing.</p>
<p>Instead, we will use <code>Picard</code> to compute the metrics:</p>
<div class="codehilite"><pre><span></span><code><span class="k">for</span> <span class="n">i</span> <span class="nb">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="nb">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">PICARD_JAR</span>} <span class="n">CollectAlignmentSummaryMetrics</span> \
    <span class="n">VALIDATION_STRINGENCY</span>=<span class="n">SILENT</span> \
    <span class="n">REFERENCE_SEQUENCE</span>=${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    <span class="n">INPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">OUTPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">alignment</span>.<span class="n">tsv</span> \
    <span class="n">METRIC_ACCUMULATION_LEVEL</span>=<span class="n">LIBRARY</span>
<span class="nb">done</span>
</code></pre></div>

<p><br>
Explore the results</p>
<div class="codehilite"><pre><span></span><code>less -S alignment/normal/normal.sorted.dup.recal.metric.alignment.tsv
less -S alignment/tumour/tumour.sorted.dup.recal.metric.alignment.tsv
</code></pre></div>

<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Do you think the sample and the reference genome fit together?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Yes, 99% of the reads have been aligned.<br />
Usually, we consider:</p>
<ul>
<li>A good alignment if &gt; 85%</li>
<li>Reference assembly issues if [60-85]%</li>
<li>Probably a mismatch between sample and ref if &lt; 60 %</li>
</ul>
</details>
</div>
</div>
<hr />
<h2 id="variant-calling">Variant Calling<a class="headerlink" href="#variant-calling" title="Permanent link">&para;</a></h2>
<p>Most of SNV caller use either a Bayesian, a threshold or a t-test
approach to do the calling</p>
<p>Here we will try 3 variant callers:</p>
<ul>
<li><code>Varscan 2</code></li>
<li><code>MuTecT</code></li>
<li><code>Strelka</code></li>
</ul>
<p>Other candidates:</p>
<ul>
<li><code>Virmid</code></li>
<li><code>Somatic sniper</code></li>
</ul>
<p>Many, MANY others can be found here: <a href="https://www.biostars.org/p/19104/">https://www.biostars.org/p/19104/</a></p>
<p>In our case, let’s create a new work directory to start with (from base
directory):</p>
<div class="codehilite"><pre><span></span><code><span class="n">cd</span><span class="w"> </span><span class="o">$</span><span class="n">SNV_BASE</span>
<span class="n">mkdir</span><span class="w"> </span><span class="n">variant_calling</span>
</code></pre></div>

<h3 id="varscan-2">Varscan 2<a class="headerlink" href="#varscan-2" title="Permanent link">&para;</a></h3>
<p><code>Varscan 2</code> calls somatic variants (SNPs and indels) using a
heuristic method and a statistical test based on the number of aligned
reads supporting each allele. It expects both a normal and a tumour file in
<code>SAMtools pileup</code> format from sequence alignments in binary alignment/map
(BAM) format. To build a pileup file, you will need:</p>
<ul>
<li>A SAM/BAM file (<code>*.sorted.dup.recal.bam</code>) that has been sorted using the <code>sort</code> command
of <code>samtools</code>.</li>
<li>The reference sequence (<code>human_g1k_v37.fasta</code>) to which reads were aligned, in
FASTA format.</li>
<li>The <code>samtools</code> software package.</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span>normal<span class="w"> </span>tumour
<span class="k">do</span>
samtools<span class="w"> </span>mpileup<span class="w"> </span>-L<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-B<span class="w"> </span>-q<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-f<span class="w"> </span><span class="si">${</span><span class="nv">REF</span><span class="si">}</span>/human_g1k_v37.fasta<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>-r<span class="w"> </span><span class="m">4</span>:1-10000000<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>alignment/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.sorted.dup.recal.bam<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>&gt;<span class="w"> </span>variant_calling/<span class="si">${</span><span class="nv">i</span><span class="si">}</span>.mpileup
<span class="k">done</span>
</code></pre></div>
<p>Notes on <code>samtools</code> arguments:</p>
<blockquote>
<p><strong>-L</strong>: max per-sample depth for INDEL calling [1000]<br />
<strong>-B</strong>: disable BAQ (per-Base Alignment Quality)<br />
<strong>-q</strong>: skip alignments with mapQ smaller than 1<br />
<strong>-g</strong>: generate genotype likelihoods in BCF format</p>
</blockquote>
<p><br>
  <div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">VARSCAN_JAR</span><span class="p">}</span><span class="w"> </span><span class="err">\</span>
<span class="n">somatic</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">mpileup</span><span class="w"> </span><span class="err">\</span>
<span class="n">variant_calling</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">mpileup</span><span class="w"> </span><span class="err">\</span>
<span class="n">variant_calling</span><span class="o">/</span><span class="n">varscan</span><span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">vcf</span><span class="w"> </span><span class="mi">1</span><span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">strand</span><span class="o">-</span><span class="n">filter</span><span class="w"> </span><span class="mi">1</span><span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">somatic</span><span class="o">-</span><span class="n">p</span><span class="o">-</span><span class="n">value</span><span class="w"> </span><span class="mf">0.001</span>
</code></pre></div></p>
<h3 id="mutect">MuTect<a class="headerlink" href="#mutect" title="Permanent link">&para;</a></h3>
<p>Now let’s try a different variant caller, <code>MuTect</code>.</p>
<div class="highlight"><pre><span></span><code><span class="n">$JAVA7</span><span class="w"> </span><span class="o">-</span><span class="n">Xmx2G</span><span class="w"> </span><span class="o">-</span><span class="n">jar</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">MUTECT_JAR</span><span class="p">}</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">T</span><span class="w"> </span><span class="n">MuTect</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">R</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">human_g1k_v37</span><span class="p">.</span><span class="na">fasta</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">dt</span><span class="w"> </span><span class="n">NONE</span><span class="w"> </span><span class="o">-</span><span class="n">baq</span><span class="w"> </span><span class="n">OFF</span><span class="w"> </span><span class="o">--</span><span class="n">validation_strictness</span><span class="w"> </span><span class="n">LENIENT</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">dbsnp</span><span class="w"> </span><span class="n">$</span><span class="p">{</span><span class="n">REF</span><span class="p">}</span><span class="o">/</span><span class="n">dbSnp</span><span class="o">-</span><span class="mi">138_</span><span class="n">chr4</span><span class="p">.</span><span class="na">vcf</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">input_file</span><span class="p">:</span><span class="n">normal</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">normal</span><span class="o">/</span><span class="n">normal</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">recal</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">input_file</span><span class="p">:</span><span class="n">tumor</span><span class="w"> </span><span class="n">alignment</span><span class="o">/</span><span class="n">tumour</span><span class="o">/</span><span class="n">tumour</span><span class="p">.</span><span class="na">sorted</span><span class="p">.</span><span class="na">dup</span><span class="p">.</span><span class="na">recal</span><span class="p">.</span><span class="na">bam</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">out</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">mutect</span><span class="p">.</span><span class="na">call_stats</span><span class="p">.</span><span class="na">txt</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">--</span><span class="n">coverage_file</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">mutect</span><span class="p">.</span><span class="na">wig</span><span class="p">.</span><span class="na">txt</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">pow</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">mutect</span><span class="p">.</span><span class="na">power</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">vcf</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">mutect</span><span class="p">.</span><span class="na">vcf</span><span class="w"> </span><span class="err">\</span>
<span class="w">  </span><span class="o">-</span><span class="n">L</span><span class="w"> </span><span class="mi">4</span><span class="p">:</span><span class="mi">1</span><span class="o">-</span><span class="mi">10000000</span>
</code></pre></div>
<h3 id="strelka">Strelka<a class="headerlink" href="#strelka" title="Permanent link">&para;</a></h3>
<p>And finally let’s try Illumina’s <code>Strelka</code>.</p>
<div class="highlight"><pre><span></span><code>cp ${STRELKA_HOME}/etc/strelka_config_bwa_default.ini .

sed &#39;s/isSkipDepthFilters =.*/isSkipDepthFilters = 1/g&#39; -i strelka_config_bwa_default.ini

${STRELKA_HOME}/bin/configureStrelkaWorkflow.pl \
  --normal=alignment/normal/normal.sorted.dup.recal.bam \
  --tumor=alignment/tumour/tumour.sorted.dup.recal.bam \
  --ref=${REF}/human_g1k_v37.fasta \
  --config=${SNV_BASE}/strelka_config_bwa_default.ini \
  --output-dir=variant_calling/strelka/

cd variant_calling/strelka/
make -j2
cd ../..

cp variant_calling/strelka/results/passed.somatic.snvs.vcf variant_calling/strelka.vcf
</code></pre></div>
<h3 id="comparing-variant-callers">Comparing variant callers<a class="headerlink" href="#comparing-variant-callers" title="Permanent link">&para;</a></h3>
<p>Now we have variants from all three methods. Let’s compress and index
the VCFs for future visualisation.</p>
<div class="highlight"><pre><span></span><code><span class="k">for</span><span class="w"> </span>i<span class="w"> </span><span class="k">in</span><span class="w"> </span>variant_calling/*.vcf<span class="p">;</span><span class="w"> </span><span class="k">do</span><span class="w"> </span>bgzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$i</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$i</span>.gz<span class="w"> </span><span class="p">;</span><span class="w"> </span>tabix<span class="w"> </span>-p<span class="w"> </span>vcf<span class="w"> </span><span class="nv">$i</span>.gz<span class="p">;</span><span class="w"> </span><span class="k">done</span>
</code></pre></div>
<p>Let’s look at a compressed VCF. Details on the VCF spec can be found <a href="https://vcftools.github.io/specs.html">here</a>.</p>
<div class="codehilite"><pre><span></span><code><span class="n">zless</span><span class="w"> </span><span class="o">-</span><span class="n">S</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">varscan</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</code></pre></div>

<p><br>
Fields vary from caller to caller. Some values are are almost always there:</p>
<ul>
<li>Ref vs. alt alleles</li>
<li>Variant quality (QUAL column)</li>
<li>The per-sample genotype (GT) values.</li>
</ul>
<p>Note on VCF fields:</p>
<blockquote>
<p><strong>DP</strong>: Raw read depth<br />
<strong>GT</strong>: Genotype<br />
<strong>PL</strong>: List of Phred-scaled genotype likelihoods. (min is better)<br />
<strong>DP</strong>: &ldquo;&rdquo;# high-quality bases&rdquo;<br />
<strong>SP</strong>: Phred-scaled strand bias P-value<br />
<strong>GQ</strong>: Genotype Quality  </p>
</blockquote>
<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Looking at the three vcf files, how can we detect only somatic variants?</p>
<div class="admonition success">
<details>
<summary><strong>Answer</strong></summary>
<p>Some commands to find somatic variant in the vcf file:</p>
<p>varscan:</p>
<div class="codehilite"><pre><span></span><code><span class="n">grep</span><span class="w"> </span><span class="n">SOMATIC</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">varscan</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span>
</code></pre></div>

<p>MuTecT:</p>
<div class="codehilite"><pre><span></span><code><span class="n">grep</span><span class="w"> </span><span class="o">-</span><span class="n">v</span><span class="w"> </span><span class="n">REJECT</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">mutect</span><span class="o">.</span><span class="n">vcf</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">grep</span><span class="w"> </span><span class="o">-</span><span class="n">v</span><span class="w"> </span><span class="s2">&quot;^#&quot;</span>
</code></pre></div>

<p>Strelka:</p>
<div class="codehilite"><pre><span></span><code><span class="n">grep</span><span class="w"> </span><span class="o">-</span><span class="n">v</span><span class="w"> </span><span class="s2">&quot;^#&quot;</span><span class="w"> </span><span class="n">variant_calling</span><span class="o">/</span><span class="n">strelka</span><span class="o">.</span><span class="n">vcf</span>
</code></pre></div>

</details>
</div>
</div>
<hr />
<h2 id="variant-visualisation">Variant Visualisation<a class="headerlink" href="#variant-visualisation" title="Permanent link">&para;</a></h2>
<p>The Integrative Genomics Viewer (<code>IGV</code>) is an efficient visualization tool
for interactive exploration of large genome datasets.</p>
<p>Before jumping into <code>IGV</code>, we’ll generate a track IGV that can be used to plot
coverage:</p>
<div class="codehilite"><pre><span></span><code><span class="k">for</span> <span class="n">i</span> <span class="nb">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="nb">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">jar</span> ${<span class="n">IGVTOOLS_PATH</span>}/<span class="n">igvtools</span>.<span class="n">jar</span> <span class="nb">count</span> \
    -<span class="n">f</span> <span class="nb">min</span>,<span class="nb">max</span>,<span class="n">mean</span> \
    <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span>.<span class="n">tdf</span> \
    <span class="n">b37</span>
<span class="nb">done</span>
</code></pre></div>

<p>Open <code>IGV</code></p>
<div class="codehilite"><pre><span></span><code>$IGV
</code></pre></div>

<p>Then:</p>
<ol>
<li>Choose the reference genome corresponding to those use for alignment (b37).</li>
<li>Load BAM files from the <code>alignment</code> folder (<code>tumour.sorted.dup.recal.bam</code> and <code>normal.sorted.dup.recal.bam</code>).</li>
<li>Load three VCF files (from <code>variant_calling</code> directory).</li>
</ol>
<div class="admonition note">
<p>Explore and play with the data:</p>
<ul>
<li>Find germline variants</li>
<li>Find somatic variants</li>
<li>Look around&hellip;</li>
</ul>
</div>
<hr />
<h2 id="variant-annotation">Variant Annotation<a class="headerlink" href="#variant-annotation" title="Permanent link">&para;</a></h2>
<p>Following variant calling, we end up with a VCF file of genomic
coordinates with the genotype(s) and quality information for each
variant. By itself, this information is not much use to us unless there
is a specific genomic location we are interested in. Generally, we next
want to annotate these variants to determine whether they impact any
genes and if so what is their level of impact (e.g. are they causing a
premature stop codon gain or are they likely less harmful missense
mutations).</p>
<p>The sections above have dealt with calling somatic variants from the
first 10Mb of chromosome 4. This is important in finding variants that
are unique to the tumour sample(s) and may have driven both tumour
growth and/or metastasis. An important secondary question is whether the
germline genome of the patient contains any variants that may have
contributed to the development of the initial tumour through
predisposing the patient to cancer. These variants <em>may not</em> be captured
by somatic variant analysis as their allele frequency may not change in
the tumour genome compared with the normal.</p>
<p>For this section, we will use <strong>all</strong> variants from the first 60Mb of
chromosome 5 that have been pre-generated using the GATK <code>HaplotypeCaller</code>
variant caller on both the normal and tumour genomes. The output of this
was GVCF files which were fed into GATK <code>GenotypeGVCFs</code> to produce a
merged VCF file. We will use this pre-generated file as we are primarily
interested in the annotation of variants rather than their generation.
The annotation method we will use is called <code>Variant Effect Predictor</code>
or <code>VEP</code> for short and is available from Ensembl <a href="http://ensembl.org/info/docs/tools/vep/index.html">here</a>.</p>
<p><br>
Our pre-generated VCF file is located in the <code>variants</code> folder. Let’s
have a quick look at the variants:</p>
<div class="codehilite"><pre><span></span><code><span class="n">zless</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">HC</span><span class="o">.</span><span class="n">chr5</span><span class="o">.</span><span class="mi">60</span><span class="n">Mb</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</code></pre></div>

<p>Notice how there are two genotype blocks at the end of each line for the
normal (<code>Blood</code>) and tumour (<code>liverMets</code>) samples.</p>
<p><br>
Let’s now run <code>VEP</code> on this VCF file to annotate each variant with its
impact(s) on the genome.</p>
<div class="highlight"><pre><span></span><code><span class="n">perl</span><span class="w"> </span><span class="nv">$VEP</span><span class="w"> </span><span class="o">--</span><span class="n">dir_cache</span><span class="w"> </span><span class="nv">$VEP_CACHE</span><span class="w"> </span><span class="o">-</span><span class="n">i</span><span class="w"> </span><span class="n">variants</span><span class="sr">/HC.chr5.60Mb.vcf.gz --vcf -o variants/</span><span class="n">HC</span><span class="o">.</span><span class="n">chr5</span><span class="mf">.60</span><span class="n">Mb</span><span class="o">.</span><span class="n">vep</span><span class="o">.</span><span class="n">vcf</span><span class="w"> </span><span class="o">--</span><span class="n">stats_file</span><span class="w"> </span><span class="n">variants</span><span class="sr">/HC.chr5.60Mb.vep.html --format vcf --offline -fork 4 --fasta ref/</span><span class="n">human_g1k_v37</span><span class="o">.</span><span class="n">fasta</span><span class="w"> </span><span class="o">--</span><span class="n">fields</span><span class="w"> </span><span class="n">Consequence</span><span class="p">,</span><span class="n">Codons</span><span class="p">,</span><span class="n">Amino_acids</span><span class="p">,</span><span class="n">Gene</span><span class="p">,</span><span class="n">SYMBOL</span><span class="p">,</span><span class="n">Feature</span><span class="p">,</span><span class="n">EXON</span><span class="p">,</span><span class="n">PolyPhen</span><span class="p">,</span><span class="n">SIFT</span><span class="p">,</span><span class="n">Protein_position</span><span class="p">,</span><span class="n">BIOTYPE</span><span class="w"> </span><span class="o">--</span><span class="n">species</span><span class="w"> </span><span class="n">homo_sapiens</span>
</code></pre></div>
<p><br>
<code>VEP</code> will take approximately 10 minutes to run and once it is finished
you will have a new VCF file with all of the information in the input
file but with added annotations in the INFO block. <code>VEP</code> also produces an
HTML report summarising the distribution and impact of variants
identified.</p>
<p>Once <code>VEP</code> is done running, let’s first look at the HTML report it
produced with the following command:</p>
<div class="codehilite"><pre><span></span><code><span class="n">firefox</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">HC</span><span class="o">.</span><span class="n">chr5</span><span class="o">.</span><span class="mi">60</span><span class="n">Mb</span><span class="o">.</span><span class="n">vep</span><span class="o">.</span><span class="n">html</span>
</code></pre></div>

<p>This report shows information on the <code>VEP</code> run, the number of variants,
the classes of variants detected, the variant consequences and the
distributions of variants through the genome. Close Firefox to resume
the terminal prompt.</p>
<p><br>
Now let’s look at the variant annotations that <code>VEP</code> has added to the VCF
file by focussing on a single variant. Let’s fetch the same variant from
the original VCF file and the annotated VCF file to see what has been
changed.</p>
<div class="codehilite"><pre><span></span><code><span class="n">zcat</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">HC</span><span class="o">.</span><span class="n">chr5</span><span class="o">.</span><span class="mi">60</span><span class="n">Mb</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">grep</span><span class="w"> </span><span class="s1">&#39;5\s174106\s&#39;</span>
<span class="n">grep</span><span class="w"> </span><span class="s1">&#39;5\s174106\s&#39;</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">HC</span><span class="o">.</span><span class="n">chr5</span><span class="o">.</span><span class="mi">60</span><span class="n">Mb</span><span class="o">.</span><span class="n">vep</span><span class="o">.</span><span class="n">vcf</span>
</code></pre></div>

<p><br>
These commands give us the original variant:</p>
<div class="codehilite"><pre><span></span><code><span class="mf">5</span><span class="w">   </span><span class="mf">174106</span><span class="w">  </span><span class="mf">.</span><span class="w">   </span><span class="n">G</span><span class="w">   </span><span class="n">A</span><span class="w">   </span><span class="mf">225.44</span><span class="w">  </span><span class="mf">.</span><span class="w">   </span><span class="n">AC</span><span class="o">=</span><span class="mf">2</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mf">4</span><span class="p">;</span><span class="n">BaseQRankSum</span><span class="o">=</span><span class="mf">1.22</span><span class="p">;</span><span class="n">ClippingRankSum</span><span class="o">=</span><span class="mf">0.811</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mf">21</span><span class="p">;</span><span class="n">FS</span><span class="o">=</span><span class="mf">0.000</span><span class="p">;</span><span class="n">GQ_MEAN</span><span class="o">=</span><span class="mf">127.00</span><span class="p">;</span><span class="n">GQ_STDDEV</span><span class="o">=</span><span class="mf">62.23</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mf">2</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">60.00</span><span class="p">;</span><span class="n">MQ0</span><span class="o">=</span><span class="mf">0</span><span class="p">;</span><span class="n">MQRankSum</span><span class="o">=</span><span class="mf">0.322</span><span class="p">;</span><span class="n">NCC</span><span class="o">=</span><span class="mf">0</span><span class="p">;</span><span class="n">QD</span><span class="o">=</span><span class="mf">10.74</span><span class="p">;</span><span class="kr">Read</span><span class="nb">Pos</span><span class="n">RankSum</span><span class="o">=</span><span class="mf">0.377</span><span class="p">;</span><span class="n">SOR</span><span class="o">=</span><span class="mf">0.446</span><span class="w">   </span><span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span><span class="w">  </span><span class="mf">0</span><span class="o">/</span><span class="mf">1</span><span class="p">:</span><span class="mf">7</span><span class="p">,</span><span class="mf">6</span><span class="p">:</span><span class="mf">13</span><span class="p">:</span><span class="mf">99</span><span class="p">:</span><span class="mf">171</span><span class="p">,</span><span class="mf">0</span><span class="p">,</span><span class="mf">208</span><span class="w"> </span><span class="mf">0</span><span class="o">/</span><span class="mf">1</span><span class="p">:</span><span class="mf">5</span><span class="p">,</span><span class="mf">3</span><span class="p">:</span><span class="mf">8</span><span class="p">:</span><span class="mf">83</span><span class="p">:</span><span class="mf">83</span><span class="p">,</span><span class="mf">0</span><span class="p">,</span><span class="mf">145</span>
</code></pre></div>

<p>and the same variant annotated is:</p>
<div class="codehilite"><pre><span></span><code><span class="mf">5</span><span class="w">   </span><span class="mf">174106</span><span class="w">  </span><span class="mf">.</span><span class="w">   </span><span class="n">G</span><span class="w">   </span><span class="n">A</span><span class="w">   </span><span class="mf">225.44</span><span class="w">  </span><span class="mf">.</span><span class="w">   </span><span class="n">AC</span><span class="o">=</span><span class="mf">2</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mf">4</span><span class="p">;</span><span class="n">BaseQRankSum</span><span class="o">=</span><span class="mf">1.22</span><span class="p">;</span><span class="n">ClippingRankSum</span><span class="o">=</span><span class="mf">0.811</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mf">21</span><span class="p">;</span><span class="n">FS</span><span class="o">=</span><span class="mf">0.000</span><span class="p">;</span><span class="n">GQ_MEAN</span><span class="o">=</span><span class="mf">127.00</span><span class="p">;</span><span class="n">GQ_STDDEV</span><span class="o">=</span><span class="mf">62.23</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mf">2</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">60.00</span><span class="p">;</span><span class="n">MQ0</span><span class="o">=</span><span class="mf">0</span><span class="p">;</span><span class="n">MQRankSum</span><span class="o">=</span><span class="mf">0.322</span><span class="p">;</span><span class="n">NCC</span><span class="o">=</span><span class="mf">0</span><span class="p">;</span><span class="n">QD</span><span class="o">=</span><span class="mf">10.74</span><span class="p">;</span><span class="kr">Read</span><span class="nb">Pos</span><span class="n">RankSum</span><span class="o">=</span><span class="mf">0.377</span><span class="p">;</span><span class="n">SOR</span><span class="o">=</span><span class="mf">0.446</span><span class="p">;</span><span class="n">CSQ</span><span class="o">=</span><span class="n">missense_variant</span><span class="err">|</span><span class="n">cGg</span><span class="o">/</span><span class="n">cAg</span><span class="err">|</span><span class="n">R</span><span class="o">/</span><span class="n">Q</span><span class="err">|</span><span class="n">ENSG00000153404</span><span class="err">|</span><span class="n">PLEKHG4B</span><span class="err">|</span><span class="n">ENST00000283426</span><span class="err">|</span><span class="mf">16</span><span class="o">/</span><span class="mf">18</span><span class="err">|||</span><span class="mf">1076</span><span class="err">|</span><span class="n">protein_coding</span><span class="p">,</span><span class="n">non_coding_transcript_exon_variant</span><span class="err">&amp;</span><span class="n">non_coding_transcript_variant</span><span class="err">|||</span><span class="n">ENSG00000153404</span><span class="err">|</span><span class="n">PLEKHG4B</span><span class="err">|</span><span class="n">ENST00000504041</span><span class="err">|</span><span class="mf">5</span><span class="o">/</span><span class="mf">8</span><span class="err">||||</span><span class="n">retained_intron</span><span class="w">  </span><span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span><span class="w">  </span><span class="mf">0</span><span class="o">/</span><span class="mf">1</span><span class="p">:</span><span class="mf">7</span><span class="p">,</span><span class="mf">6</span><span class="p">:</span><span class="mf">13</span><span class="p">:</span><span class="mf">99</span><span class="p">:</span><span class="mf">171</span><span class="p">,</span><span class="mf">0</span><span class="p">,</span><span class="mf">208</span><span class="w"> </span><span class="mf">0</span><span class="o">/</span><span class="mf">1</span><span class="p">:</span><span class="mf">5</span><span class="p">,</span><span class="mf">3</span><span class="p">:</span><span class="mf">8</span><span class="p">:</span><span class="mf">83</span><span class="p">:</span><span class="mf">83</span><span class="p">,</span><span class="mf">0</span><span class="p">,</span><span class="mf">145</span>
</code></pre></div>

<p><br>
You can see that <code>VEP</code> has added:</p>
<div class="highlight"><pre><span></span><code>CSQ=missense_variant|cGg/cAg|R/Q|ENSG00000153404|PLEKHG4B|ENST00000283426|16/18|||1076|protein_coding,non_coding_transcript_exon_variant&amp;non_coding_transcript_variant|||ENSG00000153404|PLEKHG4B|ENST00000504041|5/8||||retained_intron
</code></pre></div>
<p><br>
This is further composed of two annotations for this variant:
<div class="highlight"><pre><span></span><code>missense_variant|cGg/cAg|R/Q|ENSG00000153404|PLEKHG4B|ENST00000283426|16/18|||1076|protein_coding
</code></pre></div></p>
<p>and</p>
<div class="highlight"><pre><span></span><code>non_coding_transcript_exon_variant&amp;non_coding_transcript_variant|||ENSG00000153404|PLEKHG4B|ENST00000504041|5/8||||retained_intron
</code></pre></div>
<p><br>
The first of these is saying that this variant is a missense variant in
the gene PLEKHG4B for the transcript ENST00000283426 and the second that
it is also a non_coding_transcript_exon_variant in the transcript
ENST00000504041.</p>
<hr />
<h2 id="variant-filtration">Variant Filtration<a class="headerlink" href="#variant-filtration" title="Permanent link">&para;</a></h2>
<p>We now have a VCF file where each variant has been annotated with one or
more impacts for one or more genes. In a typical whole cancer genome,
you will have about 4-5 million variants, and therefore rows, in a VCF
file which takes up gigabytes of space. In our small example, we have
just 100,000 variants which is already too large to make any kind of
meaningful sense out of by just opening up the VCF file in a text
editor. We need a solution that allows us to perform intelligent queries
on our variants to search this mass of noise for the signal we are
interested in.</p>
<p>Luckily, such a free tool exists and is called <code>GEMINI</code>. <code>GEMINI</code> takes as
an input your annotated VCF file and creates a database file which it
can then query using Structured Query Language (SQL) commands. Not only
does <code>GEMINI</code> make your variants easily searchable, it also brings in many
external annotations to add more information about your variants (such
as their frequencies in international databases).</p>
<p>To get started with <code>GEMINI</code>, let’s make a database out of our annotated
VCF file.</p>
<div class="highlight"><pre><span></span><code><span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="k">local</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">gemini</span><span class="w"> </span><span class="k">load</span><span class="w"> </span><span class="o">-</span><span class="n">v</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">HC</span><span class="p">.</span><span class="n">chr5</span><span class="p">.</span><span class="mi">60</span><span class="n">Mb</span><span class="p">.</span><span class="n">vep</span><span class="p">.</span><span class="n">vcf</span><span class="w"> </span><span class="c1">--cores 4 --skip-gerp-bp --skip-cadd -t VEP variants/HC.chr5.60Mb.vep.vcf.db</span>
</code></pre></div>
<p>This will take approximately 10 minutes. You will see a few errors due
to multiallelic sites, normally these sites are decomposed and
normalized before creating the <code>GEMINI</code> database but this is outside the
scope of this workshop.</p>
<p>Once the database has been created let’s run a basic query to see what
kind of information we get out of <code>GEMINI</code>.</p>
<div class="highlight"><pre><span></span><code><span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="k">local</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">gemini</span><span class="w"> </span><span class="n">query</span><span class="w"> </span><span class="o">-</span><span class="n">q</span><span class="w"> </span><span class="ss">&quot;SELECT *, (gts).(*), (gt_types).(*), (gt_depths).(*), (gt_ref_depths).(*), (gt_alt_depths).(*), (gt_quals).(*) FROM variants LIMIT 10;&quot;</span><span class="w"> </span><span class="c1">--header variants/HC.chr5.60Mb.vep.vcf.db</span>
</code></pre></div>
<p>This will output a bunch of ordered information for your query to the
command line, this is usually saved to a TSV file and opened in a
spreadsheet as we will do for the next query. In the mean time, let’s
dissect this query to understand the syntax we need to use to filter our
variants. First, we have a SELECT statement which simply specifies that
we want to select data from the database. The following comma-separated
values are the columns that we want to output from the database, in this
case we are selecting all columns with the star character and then all
sub-columns for each sample with the other values. Then, we have a &ldquo;FROM
variants&rdquo; statement which is specifying the table within the database
that we want to fetch information from. Finally, the &ldquo;LIMIT 10&rdquo;
statement specifies that no more than 10 rows should be returned. In
summary then, we are asking for all columns for 10 rows from the table
<code>variants</code>. If you haven’t used SQL before don’t worry, the <code>GEMINI</code>
website is very helpful and provides many examples for how to query your
database.</p>
<p>Let’s now perform a more interesting query to find variants that have a
medium or high impact on a gene and are rare or not present in existing
international allele frequency databases. We will save the output of
this query to a file and open it up in a spreadsheet.</p>
<div class="highlight"><pre><span></span><code><span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="k">local</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">gemini</span><span class="w"> </span><span class="n">query</span><span class="w"> </span><span class="o">-</span><span class="n">q</span><span class="w"> </span><span class="ss">&quot;SELECT *, (gts).(*), (gt_types).(*), (gt_depths).(*), (gt_ref_depths).(*), (gt_alt_depths).(*), (gt_quals).(*) FROM variants WHERE (impact_severity = &#39;HIGH&#39; OR impact_severity = &#39;MED&#39;) AND (aaf_1kg_all &lt; 0.01 OR aaf_1kg_all is null) AND (aaf_esp_all &lt; 0.01 OR aaf_esp_all is null) AND (aaf_exac_all &lt; 0.01 OR aaf_exac_all is null);&quot;</span><span class="w"> </span><span class="c1">--header variants/HC.chr5.60Mb.vep.vcf.db &gt; variants/gemini-result.tsv</span>
</code></pre></div>
<p>Notice that we have added a WHERE statement which restricts the rows
that are returned based on values that we specify for specific columns.
Here, we are asking to return variants where their impact on the gene
(impact_severity column) is medium or high and the allele frequency in
1000Genomes, ESP and EXaC is less than 1% or the variant is not present
in any of these databases.</p>
<p>Now let’s open the result in a spreadsheet to look at the annotations:</p>
<div class="codehilite"><pre><span></span><code><span class="n">libreoffice</span><span class="w"> </span><span class="o">--</span><span class="n">calc</span><span class="w"> </span><span class="n">variants</span><span class="o">/</span><span class="n">gemini</span><span class="o">-</span><span class="n">result</span><span class="o">.</span><span class="n">tsv</span>
</code></pre></div>

<p>Tick the <code>Tab</code> under <code>Separated by</code> on the dialog window that comes up.  </p>
<p>You can see that the first 14 columns contain information on the variant
including its location, ref, alt, dbSNP ID, quality and type. Slowly
scroll to the right and look at the columns of data that are provided.
Most importantly, column BD includes the gene this variant impacts, BN
the impact itself and BP the impact severity. Scroll towards the end of
the spreadsheet until you get to columns ED and EE, these contain the
genotype for each of the samples. Columns EH and EI contain the total
depth for each variant in each sample and the 4 following columns
contain the reference and alternate depths for each sample. Finally,
columns EN and EO contain the genotype qualities (from the GQ field in
the VCF) for each sample. As you scroll back and forth through this
spreadsheet, you will see that <code>GEMINI</code> brings in information from a
variety of sources including: OMIM, ClinVar, GERP, PolyPhen 2, SIFT,
ESP, 1000 Genomes, ExAC, ENCODE, CADD and more! We are only looking at a
small number of variants from the start of a chromosome so not many of
these annotations will be present but in a full genome database they are
incredibly useful.</p>
<p><code>GEMINI</code> allows you to filter your variants based on any column that you
see in this results file. For example, you may want all variants in a
specific gene, in which case you would simply add &ldquo;WHERE gene = ’BRCA1’&rdquo;
to your query. For complete documentation with many examples of queries,
see the GEMINI documentation <a href="http://gemini.readthedocs.org">here</a>.</p>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>Paila U, Chapman BA, Kirchner R and Quinlan AR. &ldquo;GEMINI: Integrative
    Exploration of Genetic Variation and Genome Annotations&rdquo;. PLoS
    Comput Biol, 2013, 9(7): e1003153. doi:10.1371/journal.pcbi.1003153</p>
</li>
<li>
<p>McLaren W, Pritchard B, Rios D, Chen Y, Flicek P and Cunningham F.
    &ldquo;Deriving the consequences of genomic variants with the Ensembl API
    and SNP Effect Predictor&rdquo;. Bioinformatics, 2010, 26(16):2069-70,
    doi:10.1093/bioinformatics/btq330</p>
</li>
</ol>
<hr />
<h2 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h2>
<p>This is based on an Introduction to DNA-Seq processing for cancer data
by Mathieu Bourgey, Ph.D.  </p>
<p>This tutorial is an adaptation of the one created by Louis letourneau.
Mathieu Bourgey would like to thank and acknowledge Louis for this help and for
sharing his material. The format of the tutorial has been inspired from
Mar Gonzalez Porta. He also wants to acknowledge Joel Fillon, Louis
Letrouneau (again), Francois Lefebvre, Maxime Caron and Guillaume
Bourque for the help in building these pipelines and working with all
the various datasets.</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; The authors; 2017-
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
  </body>
</html>